x/leverage/types/tx.go

```diff
@@ -65,6 +83,9 @@ func NewMsgSetCollateral(borrower sdk.AccAddress, denom string, enable bool) *Ms
 	}
 }
 
+func (msg MsgSetCollateral) Route() string { return ModuleName }
+func (msg MsgSetCollateral) Type() string  { return EventTypeSetCollateralSetting }
```

Comment:
Why did we have to implement `Route` and `Type` for these message types? IIRC, these are for the old deprecated means of handling messages (pre gRPC routing).

Reply:
If we don't implement that it fails to Marshal a NewOperationMsg at `cosmos-sdk/types/simulation/types.go:84`

Reply:
I see. That's unfortunate :(